### ISSUE TITLE
Incorrect interpretation of aff. species

### ISSUE THREAD

**Issue by andrewrodrigues** [gbif/pipelines#496]
Uncertainty in the taxonomic determination of species as expressed by aff. (affinis)  in the scientific name is being incorrectly interpreted as the species e.g. Lepetodrilus aff. schrolli L.Beck Lepetodrilus schrolli L.Beck https://www.gbif-uat.org/occurrence/2894750054.  This species can only be interpreted to the genus level and should not be interpreted at the species level.  Another example where there should not be interpretation at the species level is when there is the qualifier cf. in the scientific name e.g. Lepetodrilus cf. schrolli L.Beck

**Comment by mdoering** [comment]
There is some uncertainty, but it is probably the species. Thats why the species name is given. Is the genus really what we want to tie it to? An unvertain (issue?) flag might be more appropriate.

**Comment by andrewrodrigues** [comment]
I picked this up through the red list category where this critically endangered species is listed as being native to Papua New Guinea https://www.iucnredlist.org/species/157722768/157733579 but this record is from Micronesia.  There is no probability that it is the species, there is an affinity to the species but not identity to it https://www.tandfonline.com/doi/abs/10.1080/02724634.1986.10011613?journalCode=ujvp20.  Interpreting to the the species does not reflect the taxonomic treatment of the species

**Comment by dschigel** [comment]
Alternative to genus-only interepretation of affs is to list aff versions next to the non-affs as equal level, but clearly different species. Technically, aff. names are provisional names and fundamentally are not different from BOLD BINs or UNITE SHs, all these are cases "where no good / better Latin name can be given" yet we know what they can be close to (cf.) or close but not equal to (aff.)

Example of such list:

- Lepetodrilus schrolli L.Beck
- Lepetodrilus aff. schrolli L.Beck 

(each with own occurrences etc.)

**Comment by mdoering** [comment]
Do you propose to create for each species in the backbone a new aff. placeholder? We can't know in advance which occurrences carry a aff or cf marker. 

I am used to `cf.` indicating an uncertain determination, e.g.  because key characters are not visible. Not a potentially new species. To represent uncertain determinations we would even want to be able to express a determination to a custom set of species or species aggregates. Dyntaxa treats them in a nice, pragmatic way:

> Arbitrary groups are quite common in Dyntaxa today. Such groups are set up as valid taxon concepts where taxon category is set to e.g. ‘Organism group’, Species complex’ or ‘Collective taxon’. All the taxon concepts which are members of such a specific group are then connected to this group hierarchically as child taxa.

**Comment by andrewrodrigues** [comment]
Both cf. and aff. should have separate placeholders to identify them as different taxonomic concepts, however, I do not recommend clustering them with the species they are affiliated with as I would not consider them necessarily the same species or species group.  Phylogeny would strengthen the argument for species groupings/complexes but in the absence of these it is difficult to know how closely related and so do not recommend clustering of these with other species.

**Comment by mdoering** [comment]
If you treat cf or aff as separate concepts you need to treat each usage of them as a separate concept, not all usages of the same name string. There is no guarantee at all that author A used Aus cf. bus in the same sense as author B.

**Comment by timrobertson100** [comment]
For info, we have:

- 157,749 occurrences with _af._ in them that get organised to (sub)species. These occurrences contain 2,612 distinct verbatim names 
- 268,140 occurrences with _cf._ in them that are organised to (sub)species. Within occurrences contain 43,839 distinct verbatim names

**Comment by andrewrodrigues** [comment]
Agreed Markus.  Then the most logical thinsg to do would be to not resolve any of the cf and aff occurrences to species level and avoid interpretations of taxonomic equivalence by us to either related species or to taxonomic determinations between collectors, and rather clump them under the genus with a taxonomy undetermined flag.

**Comment by dschigel** [comment]
If the informational loss of Aus cf. bus < Aus sp. is an acceptable loss, it sounds like a working compromise. Number shown by @timrobertson100 look a bit high to me. Agree that there is no guarantee that Authors A and B mean the same thing in their independent use of cfs. and affs. To some extended this is also true for the good names on the labels, unless name use context is provided.

**Comment by mdoering** [comment]
Should the GBIF name matching service then simply return a higher taxon match for all names with cf. or aff. in them? Or do we need a way to also flag the presence of cf. and aff. on the occurrence?

**Comment by MattBlissett** [comment]
I think we can just return a higher taxon match.  The original name remains in the verbatim fields, and is shown in search results and on occurrence pages (in orange), and included in downloads.

**Comment by mdoering** [comment]
@MattBlissett for coding aff. and cf. names should already be flagged as NameType=INFORMAL

**Comment by andrewrodrigues** [comment]
Yes to higher taxon match.

**Comment by dschigel** [comment]
Possible warning: examples where cfs. should not be lost, see OTUs here https://www.gbif.org/species/5238959. These should stay as SH1502199.08FU (cf. Tomentella badia) **not** Tomentella sp. Perhaps thousands of cfs reported by @timrobertson100 above come from UNITE mainly? Let's not loose those please, do you agree @thomasstjerne?

**Comment by thomasstjerne** [comment]
@dschigel those cf.s are constructed in the name formatting, so they will not be affected

**Comment by thomasstjerne** [comment]
@mdoering in the match service, would it be possible to infer https://dwc.tdwg.org/terms/#identificationQualifier and give it back as part of the response? And probably with a type being either `CF` or `AFF`.

Then users can choose to filter out data marked `cf` or `aff.`

> I am used to cf. indicating an uncertain determination, e.g. because key characters are not visible. 

This is also how I have learned to use`cf`. Often used in species complexes where you need to have very good material and maybe do sequencing or extensive microscopy to be 100% certain.

But `aff` means it is NOT that species, but it seems to be morphologically close. 

So I would expect to get `cf` matched to species but higher taxon match is better for `aff`.

**Comment by mdoering** [comment]
@thomasstjerne exactly as I would expect too. 
Returning identificationQualifier as part of the response sounds like a good way forward. I might need to modify the name parser for this, but that should be possible. We want controlled values for the return?

**Comment by mdoering** [comment]
Ah, Id need to change the API classes, thats a bit of work

**Comment by mdoering** [comment]
I assume the matching service should then also consume dwc:identificationQualifier as an external parameter if the occurrence has given that? If that says aff. we should do a higher match too.

**Comment by mdoering** [comment]
aff. or cf. can also be given for subspecies or I guess even for uninomials like a genus or subgenus?
I think its still good enough to just return identificationQualifier = aff. for all these names and their higher match:
 - Aus aff. bus  => Aus
 - Aus bus aff. cus  => Aus bus
 - aff. Aus => next higher rank from classification, e.g. the family

**Comment by mdoering** [comment]
@MattBlissett the native name parser expose an [epithetQualifier](https://github.com/gbif/name-parser/blob/master/name-parser-api/src/main/java/org/gbif/nameparser/api/ParsedName.java#L105
) which can be used to spot aff. or cf. and know to which part of the name it belongs to. 

The NameParserGbifV1 converts the native ParsedName into a GBIF API version, thereby losing this qualifier.
I would be glad if we could ditch the entire name-parser-v1 package and instead use the native name parser in CLB, probably keeping the conversion routine but expanding the ParsedName class from the GBIF API to also have the epithetQualifier.
